nixos/modules/services/misc/mautrix-telegram.nix

```diff
@@ -0,0 +1,155 @@
+              formatter = "precise";
+            };
+
+            loggers = {
+              mau.level = "INFO";
+              telethon.level = "INFO";
+              aiohttp.level = "INFO";
+            };
+
+            root = {
+              level = "INFO";
+              handlers = [ "console" ];
+            };
+          };
+        };
+        example = literalExample ''
+          {
+            homeserver = {
+              address = "http://localhost:8008";
+              domain = "public-domain.tld";
+            };
+
+            appservice.public = {
+              prefix = "/public";
+              external = "https://public-appservice-address/public";
+            };
+
+            bridge.permissions = {
+              "example.com" = "full";
+              "@admin:example.com" = "admin";
+            };
+          }
+        '';
+        description = ''
+          <filename>config.yaml</filename> configuration as a Nix attribute set.
+          Configuration options should match those described in
+          <link xlink:href="https://github.com/tulir/mautrix-telegram/blob/master/example-config.yaml">
+          example-config.yaml</link>.
+        '';
+        # TODO: uncomment this once mautrix-telegram v0.5.3 is out (https://github.com/tulir/mautrix-telegram/pull/332)
+        #
+        #  Secret tokens should be specified using <option>environmentFile</option>
+        #  instead of this world-readable attribute set.
+        #'';
+      };
+
+      # TODO: uncomment this once mautrix-telegram v0.5.3 is out (https://github.com/tulir/mautrix-telegram/pull/332)
+      #environmentFile = mkOption {
+      #  type = types.nullOr types.path;
+      #  default = null;
+      #  description = ''
+      #    File containing environment variables to be passed to the mautrix-telegram service,
+      #    in which secret tokens can be specified securely by defining values for
+      #    <literal>MAUTRIX_TELEGRAM_TELEGRAM_API_ID</literal>,
+      #    <literal>MAUTRIX_TELEGRAM_TELEGRAM_API_HASH</literal> and optionally
+      #    <literal>MAUTRIX_TELEGRAM_TELEGRAM_BOT_TOKEN</literal>.
+      #  '';
+      #};
+
+      homeserverService = mkOption {
+        type = types.nullOr types.str;
+        default = "matrix-synapse.service";
+        description = ''
+          Matrix homeserver service to wait for when starting the application service.
+        '';
+      };
+    };
+  };
+
+  config = let
+    cfg = config.services.mautrix-telegram;
+    overrideConfigFile = pkgs.writeText "mautrix-telegram-config.json" (builtins.toJSON cfg.config);
+    serviceConfig = pkgs.runCommand "mautrix-telegram-config" { preferLocalBuild = true; } ''
+      mkdir -p "$out"
+
+      # the configuration file is rewritten to include the registration tokens
+      cp '${overrideConfigFile}' "$out/config.yaml"
+      chmod +w "$out/config.yaml"
+
+      ${pkgs.buildPackages.mautrix-telegram}/bin/mautrix-telegram \
+        --generate-registration \
```

Comment:
What does this really do and why is it needed? It seems that this generates a new registration token every time this runs, which doesn't seem like something you'd want.

Reply:
Comment added.

I think we should completely drop the registration file generation (https://github.com/NixOS/nixpkgs/pull/63589#discussion_r299776399).

Reply:
Registration file generation removed (https://github.com/NixOS/nixpkgs/pull/63589#discussion_r300739042).